### PR TITLE
LPD-41243

### PR DESCRIPTION
**Pull request by brianikim**
https://liferay.atlassian.net/browse/LPD-41243

cc @smottal

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
@smottal , tied to LPP.

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java [hunk 1]
```diff
@@ -120,9 +128,76 @@ public void testAddUserWithInvalidName() {
 		}
 	}
 
+	@Test
+	public void testGetRoleUserIdsWithNoPermissions() throws Exception {
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		User user1 = UserTestUtil.addUser();
```
Comment (smottal) [inline modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java:136]: user1 -> user (we don't have any other users here)

File: modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java [hunk 2]
```diff
@@ -120,9 +128,76 @@ public void testAddUserWithInvalidName() {
...
+
+		try {
+			Role role = RoleTestUtil.addRole(RoleConstants.TYPE_REGULAR);
+
+			_userLocalService.addRoleUser(role.getRoleId(), role.getUserId());
```
Comment (smottal) [inline modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java:141]: mmm... don't we have to add the role to user1?

File: modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java [hunk 3]
```diff
@@ -120,9 +128,76 @@ public void testAddUserWithInvalidName() {
...
+				originalPermissionChecker);
+		}
+	}
+
+	@Test
+	public void testGetRoleUserIdsWithPermissions() throws Exception {
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		User user1 = UserTestUtil.addUser();
```
Comment (smottal) [inline modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java:169]: just user

File: modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java [hunk 4]
```diff
@@ -120,9 +128,76 @@ public void testAddUserWithInvalidName() {
...
+
+		try {
+			Role role = RoleTestUtil.addRole(RoleConstants.TYPE_REGULAR);
+
+			RoleTestUtil.addResourcePermission(
+				role, Role.class.getName(), ResourceConstants.SCOPE_COMPANY,
+				String.valueOf(TestPropsValues.getCompanyId()),
+				ActionKeys.ASSIGN_MEMBERS);
+
+			_userLocalService.addRoleUser(role.getRoleId(), user1.getUserId());
```
Comment (smottal) [inline modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java:179]: here it is correct

**Review by smottal — CHANGES_REQUESTED** [review]

File: modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java (hunk 2 above)
Comment (brianikim) [reply modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java:141]: @smottal , it was small detail I missed :sweat_smile: . I messed up when separating the test into two tests (noPermissions vs. permissions) when I was originally reusing the patterns for each method.

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:sf

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:report:65797843

**Review by smottal — APPROVED** [review]

**Comment by smottal** [comment]
hey @brianikim could you please check the report? Because there are some failing tests that I don't know if are related or not. Thx

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
@smottal , rerunning CI one more time. The Account* tests do not look related, but I want to make sure.

**Comment by smottal** [comment]
ci:report:66314882

**Comment by smottal** [comment]
ci:forward:force

**Pull request closed without merging**